### PR TITLE
Ben/long covid

### PR DESCRIPTION
**Pull request by bionara**
# Description

WIP for long covid FE. 

BE: https://github.com/zoe/covid-app/pull/478 
Requires some UX finesse, and store "completed" status in redux to prevent future BE checks

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots
image

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevant
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/features/long-covid/screens/consts.questions.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    { value: 'YES_SOME_BETTER', label: i18n.t('long-covid.symptoms-change-a2') },
+    { value: 'NO_CHANGE', label: i18n.t('long-covid.symptoms-change-a3') },
+    { value: 'YES_SOME_WORSE', label: i18n.t('long-covid.symptoms-change-a4') },
+    { value: 'YES_ALL_WORSE', label: i18n.t('long-covid.symptoms-change-a5') },
+    { value: 'SOME_IMPROVED', label: i18n.t('long-covid.symptoms-change-a6') },
+    { value: 'NOT_YET_TWO_WEEKS', label: i18n.t('long-covid.symptoms-change-a7') },
+    { value: 'OTHER', label: i18n.t('long-covid.symptoms-change-a8') },
+];
+
+export const symtpomChangesKeyList: string[] = [
```
Comment (aikewoody) [inline src/features/long-covid/screens/consts.questions.tsx:140]: Typo :)

Comment (bionara) [reply src/features/long-covid/screens/consts.questions.tsx:140]: Cheers - fixed (symptomChangesKeyList)

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        { props.values.other ? (
+          <GenericTextField
+            formikProps={props}
+            name="other"
+          />
+        ) : null }
+    </CheckboxList>
+  </View>;
+
+  const renderBulletLine = (text: string) => <View style={{ flexDirection: 'row', paddingTop: 16, paddingRight: 16, }}>
```
Comment (aikewoody) [inline src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:55]: These components should be defined outside the main render function

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            onChangeText={props.handleChange('symptom_change_comments')}
+            underline={false}
+        />
+
+    </View>
+  ): null;
+
+  return (
+    <Formik
+        style={{ margin: 16 }}
```
Comment (aikewoody) [inline src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:160]: For now it's fine that we use magic numbers like this but after deciding on a theming implementation we should stop doing this

File: src/features/long-covid/screens/LongCovidStart.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+export default function LongCovidStartScreen({ route }: IProps) {
+  //const LongCovidState = useSelector(LongCovidStart);
+  const dispatch = useDispatch();
+  const { patientData } = route.params;
+
+  console.log('START IT WITH ', patientData.patientId)
+
+  useEffect(() => {
```
Comment (aikewoody) [inline src/features/long-covid/screens/LongCovidStart.tsx:23]: It's better to comment out this obsolete useEffect or delete the whole functionality.

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx (hunk 3 above)
Comment (bionara) [reply src/features/long-covid/screens/LongCovidQuestionPageOne.tsx]: Absolutely

File: src/features/long-covid/screens/LongCovidStart.tsx (hunk 4 above)
Comment (bionara) [reply src/features/long-covid/screens/LongCovidStart.tsx]: Cleaned all this up.

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+          >
+            {i18n.t(`long-covid.q${index + 4}`)}
+          </CheckboxItem>
+        ))}
+        {props.values.other ? <GenericTextField formikProps={props} name="other" /> : null}
+      </CheckboxList>
+    </View>
+  );
+
+  const renderBulletLine = (text: string) => (
```
Comment (aikewoody) [inline src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:72]: It's better to make a separate component for this to make the code more DRY in the future, at least it shouldn't be defined in the render function.

**Review by aikewoody — APPROVED** [review]

File: src/core/state/long-covid/slice/index.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { longCovidQuestionPageOneDataInitialState } from '@covid/features/long-covid/screens/consts.questions';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from '../../root';
+import { ILongCovid } from '../types';
+
+const initialState: ILongCovid = longCovidQuestionPageOneDataInitialState;
+
+const longCovidSlice = createSlice({
```
Comment (selvach) [inline src/core/state/long-covid/slice/index.ts:9]: Does this work for multiple profiles?

File: src/features/long-covid/LongCovidApiClient.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  add(patientId: string, longCovid: ILongCovid): Promise<ILongCovid> {
+    longCovid = {
+      ...longCovid,
+      patient: patientId,
+    };
+    return this.apiClient.post<ILongCovid, ILongCovid>(API_URL, longCovid);
+  }
+
+  update(LongCovid: ILongCovid): Promise<ILongCovid> {
```
Comment (selvach) [inline src/features/long-covid/LongCovidApiClient.ts:31]: lower case variables for consistency? (Fix in later PR)

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+  const renderFormCheckboxes = (props: FormikProps<ILongCovid>) => (
+    <View style={{ marginVertical: 16 }}>
+      <CheckboxList>
+        {checkBoxQuestions4To17.map((key: string, index: number) => (
+          <View style={{ marginBottom: 16 }}><CheckboxItem
+            onChange={(value: boolean) => props.setFieldValue(key, !props.values[key])}
+            value={props.values[key]}
+            dark={true}
+          >
+            {i18n.t(`long-covid.q${index + 4}`)}
```
Comment (selvach) [inline src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:64]: Can you put the 4 as a defined constant (const NUMBER_OF_CHECKBOXES = 5) or something similar.

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+  },
+  infoBox: {
+    backgroundColor: colors.white,
+    borderRadius: 8,
+    marginTop: 16,
+    padding: 16,
+    textAlign: 'left',
+  },
+  textarea: {
+    backgroundColor: '#EEEEEF',
```
Comment (selvach) [inline src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:250]: TODO: Add colour to a const file. (Later PR is fine).

File: assets/icons/svgIcons/ClockIcon.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { G, Path, Svg } from 'react-native-svg';
+
+interface IProps {
+  color?: string;
+}
+
+function ClockIcon({ color = '#024364' }: IProps) {
```
Comment (selvach) [inline assets/icons/svgIcons/ClockIcon.tsx:8]: Add color hash code to a const file. (Can be a future PR).

**Review by selvach — APPROVED** [review]
I think some of the copy and design can be imporved. But lets get this v1 into QA and then iterate. 

Nice job - there's a lot going on!!!

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx (hunk 8 above)
Comment (bionara) [reply src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:64]: Yep!

**Comment by bionara** [comment]
> I think some of the copy and design can be imporved. But lets get this v1 into QA and then iterate.

yep, main thing is to check multiple profiles as per your comment. Defo a lot more going on than it looked!

File: src/features/long-covid/LongCovidApiClient.ts (hunk 7 above)
Comment (bionara) [reply src/features/long-covid/LongCovidApiClient.ts:31]: Done

File: src/features/long-covid/screens/LongCovidQuestionPageOne.tsx (hunk 8 above)
Comment (bionara) [reply src/features/long-covid/screens/LongCovidQuestionPageOne.tsx:64]: Done

File: assets/icons/svgIcons/ClockIcon.tsx (hunk 10 above)
Comment (bionara) [reply assets/icons/svgIcons/ClockIcon.tsx:8]: Done - we had it already, whoops!

**Pull request merged**